### PR TITLE
[Swift 4.2] Update Karatsuba Multiplication

### PR DESCRIPTION
**Pull request by afonsograca**
### Checklist

- [x] I've looked at the [contribution guidelines](https://github.com/raywenderlich/swift-algorithm-club/blob/master/.github/CONTRIBUTING.md).
- [x] This pull request is complete and ready for review.

### Description

 
Changed the way the characters in a String were accessed since `characters` was deprecated on Swift 4.0.
Removed the code snippet at the beggining of the file.
Updated the `README.md` for the algorithm.
Added a test project with a test based on the example in the `README.md`.

### CODE REVIEW

**Comment by SpacyRicochet** [comment]
References #748. Makes it easier to find algorithms to update.

File: Karatsuba Multiplication/Tests/KaratsubaMultiplicationTests.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//
+//  KaratsubaMultiplicationTests.swift
+//  Tests
+//
+//  Created by Afonso Graça on 8/10/18.
+//
+
+import XCTest
+
+final class KaratsubaMultiplicationTests: XCTestCase {
```
Comment (richard-ash) [inline Karatsuba Multiplication/Tests/KaratsubaMultiplicationTests.swift]: Awesome!! Thanks for adding tests 😃

File: Karatsuba Multiplication/README.markdown [hunk 2]
```diff
@@ -71,15 +71,15 @@ Here's the full implementation. Note that the recursive algorithm is most effici
...
-  let num1Array = String(num1).characters
-  let num2Array = String(num2).characters
+  func karatsuba(_ num1: Int, by num2: Int) -> Int {
+  let num1String = String(num1)
+  let num2String = String(num2)
   
-  guard num1Array.count > 1 && num2Array.count > 1 else {
-    return num1*num2
+  guard num1String.count > 1 && num2String.count > 1 else {
+  return multiply(num1, by: num2)
```
Comment (richard-ash) [inline Karatsuba Multiplication/README.markdown]: Indent this line 2 spaces

File: Karatsuba Multiplication/README.markdown [hunk 3]
```diff
@@ -71,15 +71,15 @@ Here's the full implementation. Note that the recursive algorithm is most effici
 
 ```swift
 // Karatsuba Multiplication
-func karatsuba(_ num1: Int, by num2: Int) -> Int {
-  let num1Array = String(num1).characters
-  let num2Array = String(num2).characters
+  func karatsuba(_ num1: Int, by num2: Int) -> Int {
```
Comment (richard-ash) [inline Karatsuba Multiplication/README.markdown]: Remove the indentation here

**Pull request merged**